futaba/journal/impl/direct_message.py

```diff
@@ -0,0 +1,58 @@
+#
+# journal/impl/direct_message.py
+#
+# futaba - A Discord Mod bot for the Programming server
+# Copyright (c) 2017-2018 Jake Richardson, Ammon Smith, jackylam5
+#
+# futaba is available free of charge under the terms of the MIT
+# License. You are free to redistribute and/or modify it under those
+# terms. It is distributed in the hopes that it will be useful, but
+# WITHOUT ANY WARRANTY. See the LICENSE file for more details.
+#
+
+"""
+A Listener that DMs messages to the configured Discord user.
+"""
+
+import logging
+
+from futaba.permissions import mod_perm
+from futaba.utils import copy_discord_file
+from ..listener import Listener
+
+logger = logging.getLogger(__name__)
+
+__all__ = ["DirectMessageListener"]
+
+
+class DirectMessageListener(Listener):
+    def __init__(self, router, path, user, recursive=True):
+        super().__init__(router, path, recursive)
+        self.user = user
+
+    async def handle(self, path, guild, content, attributes):
+        """
+        Send the message to the given channel, applying the icon if applicable.
+        """
+
+        if guild is not None:
+            content = f"**[{guild.name}]** {content}"
+
+            # Don't send journal events if they're not a mod
+            member = guild.get_member(self.user.id)
+            if member is None or not member.guild_permissions.manage_channels:
```

Comment:
Is this where `mod_perm` was to be used?
If so it should be as if we want to change the permission needed for mod we wouldn't need to change this line

Reply:
Correct: `*_perm` takes a `Context`, not a `Member`. There wasn't any reasonable way to produce one in this circumstance.

Do you have a suggestion of a more portable way to detect if a member is a moderator?